### PR TITLE
SASS-7075: Fix session for other summary journeys

### PR DESCRIPTION
**Pull request by lukastymo**

### CODE REVIEW

File: test/services/PensionSessionServiceSpec.scala [hunk 1]
```diff
@@ -128,4 +130,50 @@ class PensionSessionServiceSpec extends UnitTest with MockPensionUserDataReposit
...
+
+      val response = service.mergePriorDataToSession(taxYear, user, (_, _, _) => Ok)
+
+      assert(status(response) === OK)
+    }
+
+    "return an internal server error if data not found" in {
+      super[MockIncomeTaxUserDataConnector].mockFind(nino, taxYear, noPriorData)
+      super[MockPensionUserDataRepository].mockFind(taxYear, user, Left(DataNotFound))
+      mockCreateOrUpdate(Right(()))
```
Comment (harry-ritchie) [inline test/services/PensionSessionServiceSpec.scala:161]: I don't think we need this mock call here.

File: app/services/PensionSessionService.scala [hunk 2]
```diff
@@ -118,4 +119,32 @@ class PensionSessionService @Inject() (pensionUserDataRepository: PensionsUserDa
...
+      taxYear: Int,
+      user: User,
+      renderView: (Int, PensionsCYAModel, Option[AllPensionsData]) => Result
+  )(implicit request: Request[_], hc: HeaderCarrier, clock: Clock): Future[Result] =
+    getAndHandle(taxYear, user) { (sessionData, priorData) =>
+      createOrUpdateSessionIfNeeded(sessionData, priorData, taxYear, user, renderView)
+    }
+
+  private def createOrUpdateSessionIfNeeded(
+      pensionsUserData: Option[PensionsUserData],
```
Comment (harry-ritchie) [inline app/services/PensionSessionService.scala:133]: Minor thing, but I think it helps with readability to rename `pensionsUserData` to `sessionData` so it is obvious at first glance what it is.

File: it/controllers/pensions/incomeFromPensions/IncomeFromPensionsSummaryControllerISpec.scala [hunk 3]
```diff
@@ -265,12 +265,30 @@ class IncomeFromPensionsSummaryControllerISpec extends CommonUtils with BeforeAn
...
+          lazy val result: WSResponse = showPageNoData(user)
 
-          "has an SEE_OTHER status" in {
-            result.status shouldBe SEE_OTHER
-            result.header("location") shouldBe Some(pensionSummaryUrl(taxYearEOY))
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+            println("Here db should have empty document")
```
Comment (harry-ritchie) [inline it/controllers/pensions/incomeFromPensions/IncomeFromPensionsSummaryControllerISpec.scala:275]: Do we need this println?

Comment (lukastymo) [reply it/controllers/pensions/incomeFromPensions/IncomeFromPensionsSummaryControllerISpec.scala:275]: removed

File: test/services/PensionSessionServiceSpec.scala (hunk 1 above)
Comment (lukastymo) [reply test/services/PensionSessionServiceSpec.scala:161]: yes, removed

File: app/services/PensionSessionService.scala (hunk 2 above)
Comment (lukastymo) [reply app/services/PensionSessionService.scala:133]: done

**Review by harry-ritchie — APPROVED** [review]

**Review by harry-ritchie — DISMISSED** [review]

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**